tests/unit/test_index.py

```diff
@@ -153,7 +61,9 @@ def test_get_metadata_yaml_retrieve_discourse_error(tmp_path: Path):
     assert: then ServerError is raised.
     """
     create_metadata_yaml(
-        content=f"{index.METADATA_DOCS_KEY}: http://server/index-page", path=tmp_path
+        content=f"{metadata.METADATA_NAME_KEY}: name\n"
```

Comment:
This indicates a potential refactor of the `index` module since the handling of the `metadata.yaml` file should be in that module, the `index` tests should no longer require creating the file. An option would be passing the metadata in for `index` rather than reading it. What do you think?

Reply:
Yup! I was thinking of pushing those changes with the next PR on migration when the control flow requires moving the handling of metadata.yaml up the flow :) Would that be okay? Otherwise I don't mind pushing those changes with this PR but I thought it would make more sense within the context of migration

Reply:
Yup! change on its way! 👍